src/helpers/privacy_budget.rs

```diff
@@ -0,0 +1,202 @@
+use crate::threshold::Ciphertext;
+use hex::encode as hex;
+/// This module implements privacy budget using a redis cache
+use redis::Commands;
+
+struct PrivacyBudget {
+    redis_connection: redis::Connection,
+    total_privacy_budget: i32,
+}
+
+trait ToString {
+    fn to_string(&self) -> String;
+}
+
+impl ToString for Ciphertext {
+    fn to_string(&self) -> String {
+        let (c0, c1) = self.inner();
+        let mut c0_hex = hex(c0.compress().to_bytes());
+        let c1_hex = hex(c1.compress().to_bytes());
+        c0_hex.push_str(&c1_hex);
+        c0_hex
+    }
+}
+impl PrivacyBudget {
+    fn connect(
+        redis_host_name: String,
+        redis_password: Option<String>,
+        is_tls: Option<bool>,
+    ) -> PrivacyBudget {
+        //if Redis server needs secure connection
+        let uri_scheme: String = if is_tls.unwrap_or_default() == true {
+            "rediss".to_string()
+        } else {
+            "redis".to_string()
+        };
+        let redis_conn_url = format!(
+            "{}://:{}@{}",
+            uri_scheme,
+            redis_password.unwrap_or_default(),
+            redis_host_name
+        );
+
+        PrivacyBudget {
+            redis_connection: redis::Client::open(redis_conn_url)
+                .expect("Invalid connection URL")
+                .get_connection()
+                .expect("failed to connect to Redis"),
+            total_privacy_budget: 100,
```

Comment:
For now have hard coded total_privacy_budget inside the module. Do you think this should be configurable?